libs/core/src/main/java/org/elasticsearch/core/AbstractRefCounted.java

```diff
@@ -125,4 +125,15 @@ public String toString() {
         };
     }
 
+    /**
+     * Constructs an {@link AbstractRefCounted} which does nothing when all references are released. It is the responsibility of the caller
+     * to run whatever release logic should be executed when {@link AbstractRefCounted#decRef()} returns true.
+     */
+    public static RefCounted plain() {
+        return new AbstractRefCounted() {
```

Comment:
Given how many static methods are on `AbstractRefCounted`, and its referenced directly, is it worth renaming the class to something like `DefaultRefCounted`, `SimpleRefCounted`, `BaseRefCounted` etc?